src/Compilers/Core/Portable/PEWriter/ContentId.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+
+namespace Microsoft.Cci
+{
+    internal struct ContentId
+    {
+        public readonly byte[] Guid;
+        public readonly byte[] Stamp;
+
+        public ContentId(byte[] guid, byte[] stamp)
+        {
+            Debug.Assert(guid.Length == 16 && stamp.Length == 4);
+
+            Guid = guid;
+            Stamp = stamp;
+        }
+
+        internal static ContentId FromSha1(ImmutableArray<byte> sha1)
+        {
+            var guid = new byte[16];
+            for (var i = 0; i < guid.Length; i++)
+            {
+                guid[i] = sha1[i];
+            }
+
+            // modify the guid data so it decodes to the form of a "random" guid ala rfc4122
```

Comment:
This looks suspiciously copy-and-paste. Please see my https://github.com/gafter/roslyn/pull/3 which moves it to a common API.